pkg/connector/datasets.go

```
@@ -213,7 +213,9 @@ func (o *datasetBuilder) Grants(ctx context.Context, resource *v2.Resource, pTok
 	dataset, err := ds.Metadata(ctx)
 	if err != nil {
 		if !isPermissionDenied(ctx, err) {
-			return nil, "", nil, wrapError(err, "Unable to fetch dataset metadata (projectId:"+projectId+" datasetID:"+datasetID+")")
+			// Error fetching dataset metadata, omit present resource and continue.
+			l.Debug("Unable to fetch dataset metadata (projectId:" + projectId + " datasetID:" + datasetID + ") error: " + err.Error())
+			return nil, "", nil, nil
```

Besides permission denied, what is the error that happens that you want to ignore? This change ignores all errors, including connection refused, internal server error, and even a rate limit error.

I really think this check should be more specific.

Also I notice that if the error is permission denied, we keep going and reference dataset.Access, which probably fails because dataset is nil. This is also true for fetching the policy below.

---

Hi geoff, thanks for taking a look. When making the call for the metadata, the client already handles rate limit if I'm not wrong (at least it retries) as well as other errors like internal server error. I'm not sure if we want to be specific about the error we ignore an continue, it depends on whether we prefer to interrupt the sync or not. The root of the ticket was a not found error. I though the idea was to continue with the sync when possible. I'll take a look at the case of permission denied error